lesson4/argirovatina_hometask.rb

```diff
@@ -0,0 +1,49 @@
+class LazyHash < Hash
+
+  def method_missing(method_name)
+    method_name = get_correct_key(method_name)
+    if self.keys.include? method_name
+      self[method_name] = self[method_name].call
+    else
+      raise "Method #{method_name} is missing"
+    end
+  end
+
+  def lazy?(key)
+    key = get_correct_key(key)
+    self[key].respond_to? :call
+  end
+
+  def respond_to_missing?(method_name, include_private = false)
+    method_name = get_correct_key(method_name)
+    self.keys.include? method_name || super
+  end
+
+  private
+  def get_correct_key(hash_key)
+    correct_key = [hash_key.to_s, hash_key.to_sym].select { |key| self[key] != nil }
+    raise "Keys #{correct_key} as string and as symbol are present" if correct_key.count > 1
+    raise "Method #{hash_key} is missing" if correct_key[0].nil?
+    correct_key[0]
+  end
+
+end
+
+########## EXAMPLES ###################
+proc = proc { |h, k| h[k] = k*k }
+proc_2 = proc { 2*2 }
+hash = LazyHash.new
+
+p hash[:proc_key] = proc # => #<Proc:0x007fb71e8db968@/Users/tina/Betlab-ruby-courses/lesson4/argirovatina_hometask.rb:32>
+p hash.lazy?(:proc_key) # => true
```

Comment:
Did you try to run your code?
It fails because your proc (which is hosted in `hash[:proc_key]`) expects to manage to number variables to multiply them.